src/java/org/apache/cassandra/utils/JVMStabilityInspector.java

```diff
@@ -186,17 +205,39 @@ public static void userFunctionTimeout(Throwable t)
         {
             case die:
                 // policy to give 250ms grace time to
-                ScheduledExecutors.nonPeriodicTasks.schedule(() -> killer.killCurrentJVM(t), 250, TimeUnit.MILLISECONDS);
+                ScheduledExecutors.nonPeriodicTasks.schedule(() -> killer.killJVM(t), 250, TimeUnit.MILLISECONDS);
                 break;
             case die_immediate:
-                killer.killCurrentJVM(t);
+                killer.killJVM(t);
                 break;
             case ignore:
                 logger.error(t.getMessage());
                 break;
         }
     }
 
+    public static void registerShutdownHook(Thread hook, Runnable runOnHookRemoved)
+    {
+        Runtime.getRuntime().addShutdownHook(hook);
+        shutdownHooks.add(Pair.create(hook, runOnHookRemoved));
+    }
+
+    public static void removeShutdownHooks()
+    {
+        Throwable err = null;
+        for (Pair<Thread, Runnable> hook : shutdownHooks)
+        {
+            err = Throwables.perform(err,
+                                     () -> Runtime.getRuntime().removeShutdownHook(hook.left),
+                                     hook.right::run);
+        }
+
+        if (err != null)
+            logger.error("Got error when removing shutdown hook(s): {}", err.getMessage(), err);
```

Comment:
Should it be?:
```suggestion
            logger.error("Got error(s) when removing shutdown hook(s): {}", err.getMessage(), err);
```

Reply:
> Is it correct to log the error if `killJVM` was called to be quiet? To me it sounds as violation of the [interface description](https://github.com/datastax/cassandra/pull/271/files#diff-eadfd87d7242f6ef077d475dd3d9a4e4240ddd843ac420d0f0d0bd9c3c25da8cR41):
> 
> > @param quiet - whether the error should be logged verbosely

In this case, I think it's ok. 
The `quiet` flag controls the verbose printing of a stack trace to syserr and log file about the error that triggered the shutdown.
However, this log message is about a secondary error while removing shutdown hooks and may be important diagnostic/debug information.
In other words, the quiet flag means shutdown without verbose reporting on the error that triggered the shutdown, but if there is a subsequent error removing shutdown hooks, those are still logged (not quite as verbosely since it goes to the log file only).
Well, in the end I guess it's subjective but imo the current arrangement is likely intended and seems ok to me.

Reply:
I agree. I should be more thoughtful here during review :)
BTW, when `quiet` is `false` the print of error contains unnecessary extra space between the sentences in [this line of the implementation](https://github.com/datastax/cassandra/pull/271/files#diff-bafdf887292cd447d5335dde9d5864be641eabc86d3f91e24754d3ab90f73377R271).

Reply:
No worries, it was a good discussion to have :)